src/database/schema.zmodel

```diff
@@ -797,3 +804,178 @@ model LastKeycloakUpdate {
+  @@allow("delete", auth().memberId == shoppable.authorId || has(auth().policies, "webshop:manage"))
+
+  @@map("item_question")
+}
+
+model ItemQuestionAnswer {
+  id String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  questionId String @db.Uuid
+  question ItemQuestion @relation(fields: [questionId], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  
+  answer String
+  answerEn String?
+  extraPrice Int?
+
+  // same access as ItemQuestion
+  @@allow("create", question.shoppable.authorId == auth().memberId)
+  @@allow("read", true)
+  @@deny("read", question.shoppable.removedAt != null && !(auth().memberId == question.shoppable.authorId || has(auth().policies, "webshop:manage")))
+  @@allow("update", auth().memberId == question.shoppable.authorId || has(auth().policies, "webshop:manage"))
+  @@allow("delete", auth().memberId == question.shoppable.authorId || has(auth().policies, "webshop:manage"))
+
+  @@map("item_question_answer")
+}
+
+model ItemQuestionResponse {
+  id String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  questionId String @db.Uuid
+  question ItemQuestion @relation(fields: [questionId], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  consumableId String @db.Uuid
+  consumable Consumable @relation(fields: [consumableId], references: [id], onDelete: Cascade, onUpdate: NoAction)
+
+  answer String
+  extraPrice Int?
+
+  // create: customer
+  @@allow("create", consumable.memberId == auth().memberId
+  || auth().externalCode != null && consumable.externalCustomerCode == auth().externalCode)
+  // read: customer, author of shoppable, anyone with manage or read_purchases
+  @@allow("read", consumable.memberId == auth().memberId 
+  || auth().externalCode != null && consumable.externalCustomerCode == auth().externalCode 
+  || question.shoppable.authorId == auth().memberId 
+  || hasSome(auth().policies, ["webshop:manage", "webshop:read_purchases"]))
+  // update: customer as long as not purchased
+  @@allow("update", consumable.purchasedAt == null && (auth().memberId == consumable.memberId || auth().externalCode != null))
+  // no one can delete
+  @@deny("delete", false)
+
+  @@map("item_question_response")
+}
+
+
+
+model Consumable {
+  id String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+
+  shoppableId String @db.Uuid
+  shoppable Shoppable @relation(fields: [shoppableId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+
+  memberId String? @db.Uuid 
+  member Member? @relation(fields: [memberId], references: [id])
+  externalCustomerEmail String? // if purchased by a non-member
+  externalCustomerCode String? // if purchased by a non-member, unique code used in link as security
+
+  createdAt DateTime @default(now()) @map("created_at") @db.Timestamptz(6)
+  purchasedAt DateTime? @map("purchased_at") @db.Timestamptz(6)
+  // customer or webshop:consume can always consume from unconsumed state. manager can "unconsume" from consumed state
```

Comment:
We talked about not having the user be able to consume consumables on their own, to prevent people from accidentally spending tickets before an event, for example. Instead it's people at the event who spend them for you. What do you think of that?

Reply:
Yes exactly, but then its not the owner who consumes it, right? It would be the host who does that

Reply:
sorry my comment might be weird. Anyways, I removed it so you can't consume it on your own.

What I meant is that I thought of having something like that, but I was also thinking that we might want to add an override feature. But that could be a feature in the future.